### ISSUE TITLE
Processor stops consuming messages after TCP error

### ISSUE THREAD

**Issue by wwaldner-amtelco** [lovoo/goka#334]
When a processor encounters an error reading from broker, processing stops and will not continue until restarted.  The processor should either recover or return an error so application can recover and not get stuck.

```
Error executing group consumer (continuing execution): kafka: error while consuming <SOME_EVENT>:
read tcp 10.0.1.141:38588->10.0.6.121:9092: i/o timeout
```

**Comment by frairon** [comment]
Hi @wwaldner-amtelco, 
we also noticed a couple of weeks ago, that there's a problem when the processor experiences an error. However, this only affects panics in the user-callback. Those should lead to a processor-stop, which they don't. 

In your case, sarama itself reports the error, which is tolerated intentionally.
As the message suggests, the consumer-group will restart into the next session (do a rebalance if necessary) and continue processing messages. Are you sure it stops processing entirely?

**Comment by wwaldner-amtelco** [comment]
Yes, I agree that Sarama is running into issue.  I am sure that it does not recover. Basically after the error occurs, no further messages are processed and I must restart my app to get things working again.

**Comment by frairon** [comment]
That sucks. I'll look into it, seems critical. Thanks for reporting!

**Comment by wwaldner-amtelco** [comment]
Agree.  Thanks for looking into this. If there is anything I can do to help, let me know.

**Comment by frairon** [comment]
Which version of goka are you using? Maybe we already fixed something (and did not release it). 
I'd like to reproduce this behavior, but triggering a `i/o timeout` isn't that easy (or is it?), so I'm a bit stuck there. The thing is, that we also experienced the the same error from time to time, but the processor was never stuck but continued to operate normally.
As a brute force approach we could add an option that configures the processor to crash on every error, regardless where it comes from.

**Comment by frairon** [comment]
@wwaldner-amtelco sorry, haven't found the time to implement anything yet. 
One more idea though: could you turn on goka/sarama debug-logging on the affected processor? 
For this, call `goka.SetDebug(true, true)`, documented [here](https://pkg.go.dev/github.com/lovoo/goka@master#Debug) if you're working on master, or `logger.SetDebug(true, true)` if you're on the latest tag.
That should enable debug logging. So after the error (or any error), you should see that it rebalances again with something like
`setup generation 123, claims=...` between the logging noise.
If that happens and it's still stuck then we know at least that the group consumer logic is fine and it appears to be a Kafka issue.
Let me know what you find out.

**Comment by frairon** [comment]
@wwaldner-amtelco, we have been working on a fix for this, and have an updated version released to (v1.1.0-beta.1)[https://github.com/lovoo/goka/releases/tag/v1.1.0-beta.1]. Take a look if you have time to check if that fixes your issue.

**Comment by frairon** [comment]
I'd consider it fixed. If you still experience issues, feel free to reopen the issue.
